banano-quest/banano-quest/Models/Quest+CoreDataClass.swift

```
@@ -0,0 +1,75 @@
+//
+//  Quest+CoreDataClass.swift
+//  banano-quest
+//
+//  Created by Pabel Nunez Landestoy on 6/26/18.
+//  Copyright © 2018 Michael O'Rourke. All rights reserved.
+//
+//
+
+import Foundation
+import CoreData
+
+@objc(Quest)
+public class Quest: NSManagedObject {
+    
+    convenience init(obj: [AnyHashable: Any]!, context: NSManagedObjectContext) {
+        self.init(context: context)
+        self.questID = getLocalQuestCount(context: context) + 1
+        self.creator = obj["creator"] as? String
+        self.name = obj["name"] as? String
+        self.hint = obj["hint"] as? String
+        self.maxWinners = (obj["maxWinners"] as? Int16) ?? 0
+        self.merkleRoot = obj["merkleRoot"] as? String
+        self.metadata = Metadata(obj: obj["metadata"] as? [AnyHashable:Any], context: context)
+        self.winners = Winners(obj: obj["metadata"] as? [AnyHashable:Any], context: context)
+    }
+    
+    func save() {
+        do {
+            try self.managedObjectContext?.save()
+        } catch let error as NSError {
+            print("Failed to save Quest with error: \(error)")
```

Let's not fail silently.